### ISSUE TITLE
tagpanel clear filter button bad accessibility

### ISSUE THREAD

**Issue by t3chguy** [vector-im/riot-web#6858]
should be disabled/gone instead of invisible when its not shown

**Comment by fuzzbomb** [comment]
To expand on this...

- As a brand new user, I had no idea what a tagpanel was, let alone what happens there.  I still haven't found out how to use it.
- Turns out the tab panel is the `div.mx_TagPanel` which is 6-px wide.
- As a brand new user, it's (visually) empty, apparently because I have no communities.

The button I found was this:

`<div class="mx_TagPanel_clearButton mx_AccessibleButton" tabindex="0" role="button"><div></div</div>`

Now this is tabbable, and has keypress and click handlers.  So it exists as a tab stop for keyboard users.  But is has no content visually, and no focus style either.  In WCAG principles, it's operable, but not perceivable, and hence not understandable.

It also has a button role, but no content semantically, and no accessible name.  So it exists as a mystery for a screen reader users too.

I've  been shown screenshots showing a big X icon there once I have communities. (Haven't managed that yet).

To fix:  If seems it's intended a placeholder for a future button, so either:

- Remove `div.mx_TagPanel_clearButton` completely, and don't create it until it has a job to do.  This is probably simplest approach.
- Or, ensure it has an accessible name, mark it as disabled semantically, and take it out of the tabbing order.  Make sure it's purpose is understood visually, including the fact that it is disabled.  Something like: `<div class="mx_TagPanel_clearButton mx_AccessibleButton" tabindex="0" role="button" aria-disable="true"><img scr="clear.png" alt="Clear communities"></div>`.

Ideally, it would be a `<button disabled class="mx_TagPanel_clearButton">` -  there's [no need to use ARIA](https://www.w3.org/TR/using-aria/#rule1) for something HTML can already do.

**Comment by lampholder** [comment]
Thank you for this write up - I think you're right: having the button not exist until it has a job to do makes the most sense :smile:
